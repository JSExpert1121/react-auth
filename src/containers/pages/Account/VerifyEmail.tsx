/**
 * @name VerifyEmail
 * @description Page for email verification
 * @version 1.0
 * @todo Design
 * @author Darko
 */
import * as React from 'react';
import { connect } from 'react-redux'
import { RouteComponentProps } from 'react-router-dom';

// material ui components & icons
import Paper from '@material-ui/core/Paper';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import LinearProgress from '@material-ui/core/LinearProgress';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Link from '@material-ui/core/Link';
import { StyledComponentProps, withStyles } from '@material-ui/core/styles';

// 3rd party libraries
import { injectIntl, IntlShape, MessageDescriptor } from 'react-intl';
import messages from 'translations/messages/page.account';
import queryString from 'querystring';

// custom components
import Logo from 'assets/images/logo_text.png';

// actions & apis
import * as MemberApis from 'service/member';
import * as GlobalActions from 'store/actions/global';
import * as UserActions from 'store/actions/user';

// constants & types
import { User } from 'types/user';
import { styleFn } from './styles';


const styles = styleFn;

interface IVerifyEmailPageProps extends RouteComponentProps, StyledComponentProps {
    intl: IntlShape;
    token: string;
    user: User;
    setTitle: (title: string) => void;
    getProfile: (token: string, id?: number) => Promise<void>;
}

interface IVerifyEmailPageState {
    isBusy: boolean;
    error?: MessageDescriptor;
}

export class VerifyEmailPage extends React.Component<IVerifyEmailPageProps, IVerifyEmailPageState> {

    constructor(props: Readonly<IVerifyEmailPageProps>) {
        super(props);

        this.state = {
            isBusy: true,
            error: undefined
        }
    }

    async componentDidMount() {
        const { location, token, getProfile, setTitle, intl } = this.props;
        const values = queryString.parse(location.search.slice(1));
        const secToken = values.secureToken as string;

        setTitle(intl.formatMessage(messages.emailTitle));
        try {
            await MemberApis.verifyEmail(token, secToken);
            await getProfile(token);
            this.setState({ isBusy: false });
        } catch (error) {
            console.log('VerifyEmailPage.CDM: ', error);
            this.setState({
                isBusy: false,
                error: messages.emailVerifyFailed
            });
        }
    }

    goDashboard = () => {
        this.props.history.replace('/user');
    }

    public render() {
        const { classes, intl } = this.props;
        const { isBusy, error } = this.state;
        return (
            <Paper component='div' className={classes.root}>
                <CssBaseline />
                <FormControl className={classes.form}>
                    <Box className={classes.logo}>
                        <img src={Logo} alt={'Hit!T'} />
                    </Box>
                    <Typography className={classes.title}>
                        {intl.formatMessage(messages.emailTitle)}
                    </Typography>
                    <LinearProgress
                        id='busy-indicator'
                        className={classes.busy}
                        style={{ visibility: isBusy ? 'visible' : 'hidden' }} />
                    {error && (
                        <Typography className={classes.error}>
                            {intl.formatMessage(error)}
                        </Typography>
                    )}
                    {!error && !isBusy && (
                        <Typography className={classes.text}>
                            {intl.formatMessage(messages.emailVerified)}
                        </Typography>
                    )}
                    {!isBusy && (
                        <Box className={classes.bottomMenu}>
                            <Link style={{ cursor: 'pointer', fontSize: '1.1rem', flex: 1, textAlign: 'left' }} onClick={this.goDashboard}>
                                {intl.formatMessage(messages.toDashboard)}
                            </Link>
                        </Box>
                    )}
                </FormControl>
            </Paper>
        );
    }
};

const mapStateToProps = state => ({
    token: state.user.token,
    user: state.user.user
})

const mapDispatchToProps = {
    setTitle: GlobalActions.setTitle,
    getProfile: UserActions.getProfile
}

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(VerifyEmailPage)));
