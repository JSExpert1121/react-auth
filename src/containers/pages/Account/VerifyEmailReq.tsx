/**
 * @name VerifyEmailReq
 * @description Page for sending email verification request
 * @version 1.0
 * @todo Design
 * @author Darko
 */
import * as React from 'react';
import { connect } from 'react-redux'
import { RouteComponentProps } from 'react-router-dom';

// material ui components & icons
import Paper from '@material-ui/core/Paper';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControl from '@material-ui/core/FormControl';
import LinearProgress from '@material-ui/core/LinearProgress';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';
import { makeStyles } from '@material-ui/core/styles';

// i18n
import { injectIntl, IntlShape } from 'react-intl';
import messages from 'translations/messages/page.account';

// custom components
import Logo from 'assets/images/logo_text.png';

// actions & apis
import * as MemberApis from 'service/member';
import * as GlobalActions from 'store/actions/global';

// constants & types
import { UserProfile, User } from 'types/user';
import { styleFn } from './styles';


const useStyles = makeStyles(styleFn);

interface IVerifyEmailPageProps extends RouteComponentProps {
    intl: IntlShape;
    profile: UserProfile;
    token: string;
    user: User;
    setTitle: (title: string) => void;
}

export const VerifyEmailReqPage: React.FunctionComponent<IVerifyEmailPageProps> = (props) => {

    const [busy, setBusy] = React.useState(false);
    const [sent, setSent] = React.useState(false);

    const classes = useStyles({});
    const { user, intl, setTitle, token, history } = props;
    React.useEffect(() => {
        setTitle(intl.formatMessage(messages.emailTitle));
    })

    const handleRequest = async () => {
        try {
            setBusy(true);
            await MemberApis.requestVerificationEmail(token);
            setBusy(false);
            setSent(true);
        } catch (error) {
            console.log('VerifyEmailReqPage.handleRequest: ', error);
            setBusy(false);
        }
    }

    const goBack = () => {
        history.goBack();
    }

    return (
        <Paper component='div' className={classes.root}>
            <CssBaseline />
            <FormControl className={classes.form}>
                <Box className={classes.logo}>
                    <img src={Logo} alt={'Hit!T'} />
                </Box>
                <Typography className={classes.title}>
                    {intl.formatMessage(messages.emailTitle)}
                </Typography>
                <LinearProgress
                    id='busy-indicator'
                    className={classes.busy}
                    style={{ visibility: busy ? 'visible' : 'hidden' }} />
                <Typography className={classes.text}>
                    {`Hello ${user.name}, you're almost ready to start enjoying Hit! T service. Simply click the below button to verify your email address`}
                </Typography>
                {!sent && (
                    <Button
                        type="submit"
                        fullWidth
                        variant="contained"
                        color="primary"
                        className={classes.submit}
                        onClick={handleRequest}
                    >
                        {`Verify your mail`}
                    </Button>
                )}
                {sent && (
                    <Typography className={classes.success}>
                        Email was sent.
                    </Typography>
                )}
                {sent && (
                    <Box className={classes.bottomMenu}>
                        <Link style={{ cursor: 'pointer', fontSize: '1.1rem', flex: 1, textAlign: 'left' }} onClick={goBack}>
                            Back to Profile
                        </Link>
                        <Link style={{ cursor: 'pointer', fontSize: '1.1rem' }} onClick={handleRequest}>
                            Resend mail
                        </Link>
                    </Box>
                )}
            </FormControl>
        </Paper>
    );
};

const mapStateToProps = state => ({
    profile: state.user.profile,
    token: state.user.token,
    user: state.user.user
})

const mapDispatchToProps = {
    setTitle: GlobalActions.setTitle,
}

export default injectIntl(connect(mapStateToProps, mapDispatchToProps)(VerifyEmailReqPage));
