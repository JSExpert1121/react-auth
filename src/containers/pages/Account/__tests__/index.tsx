import React from 'react';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16/build';

import { createShallow } from '@material-ui/core/test-utils';
import ProfilePage from '../Profile';
import SettingPage from '../Setting';
import VerifyEmailReq from '../VerifyEmailReq';
import VerifyEmail from '../VerifyEmail';
import AccountPage from '../index';

configure({ adapter: new Adapter() });

describe('containers/pages/Account routing', () => {

    const shallow = createShallow();
    const wrapper = shallow(<AccountPage match={{ url: '/' }} />);

    test('Snapshot test', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
