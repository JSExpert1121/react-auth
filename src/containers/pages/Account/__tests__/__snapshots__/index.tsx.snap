// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/Account routing Snapshot test 1`] = `ShallowWrapper {}`;
